check/dumptypes.hs

```diff
@@ -1,10 +1,48 @@
+
 module Main (main) where
 
-import           System.Environment (getArgs)
-import           TypeDump (typeDump)
+import           Data.List
+import           System.Environment
+
+import           Util
 
 main :: IO ()
 main = do
   [module_] <- getArgs
-  types <- typeDump module_
-  putStr types
+  index module_
+
+index :: String -> IO ()
+index module_ = ghci ("check-hs/" ++ module_ ++ ".check.hs") input >>= putStr . normalize
+  where
+    input = ":browse! " ++ module_
+
+normalize :: String -> String
+normalize = unlines . map removeSignature . removeTypeDefinitions . lines . normalizeSignatures . unlines . removeClasses . normalizeClasses . lines
```

Comment:
Why do we call `normalizeClasses` immediately followed by `removeClasses` here?

Reply:
This is existing definition; I haven't touched it.

Reply:
Ugh, quite right. I really ought to figure out what is going on here and document this, but sadly, I don't have the energy at the moment.